docs/src/pages/examples/memory/memory-with-pg.mdx

```diff
@@ -0,0 +1,105 @@
+# Memory with PostgreSQL
+
+This example demonstrates how to use Mastra's memory system with PostgreSQL as the storage backend.
+
+## Setup
+
+First, set up the memory system with PostgreSQL storage and vector capabilities:
+
+```typescript
+import { Memory } from '@mastra/memory';
+import { PostgresStore } from '@mastra/store-pg';
+import { PgVector } from '@mastra/vector-pg';
+import { Agent } from '@mastra/core/agent';
+
+// PostgreSQL connection details
+const host = 'localhost';
+const port = 5432;
+const user = 'postgres';
+const database = 'postgres';
+const password = 'postgres';
+const connectionString = `postgresql://${user}:${password}@${host}:${port}`;
+
+// Initialize memory with PostgreSQL storage and vector search
+const memory = new Memory({
+  storage: new PostgresStore({
+    host,
+    port,
+    user,
+    database,
+    password,
+  }),
+  vector: new PgVector(connectionString),
+  options: {
+    lastMessages: 10,
+    historySearch: {
```

Comment:
this API is making me think maybe there is a better name here too. that would clearly distinguish from `lastMessages` since history sort of sounds like last messages. Maybe something like `semanticRecall`